src/message/messagesReducer.js

```diff
@@ -15,126 +16,110 @@ import {
-  const oldMessage = state[action.message_id];
+  const oldMessage = state.get(action.message_id);
   if (!oldMessage) {
     return state;
   }
-  return {
-    ...state,
-    [action.message_id]: {
-      ...oldMessage,
-      reactions: oldMessage.reactions.concat({
-        emoji_name: action.emoji_name,
-        user_id: action.user_id,
-        reaction_type: action.reaction_type,
-        emoji_code: action.emoji_code,
-      }),
-    },
-  };
+  return state.set(action.message_id, {
+    ...oldMessage,
+    reactions: oldMessage.reactions.concat({
+      emoji_name: action.emoji_name,
+      user_id: action.user_id,
+      reaction_type: action.reaction_type,
+      emoji_code: action.emoji_code,
+    }),
+  });
 };
 
 const eventReactionRemove = (state, action) => {
-  const oldMessage = state[action.message_id];
+  const oldMessage = state.get(action.message_id);
   if (!oldMessage) {
     return state;
   }
-  return {
-    ...state,
-    [action.message_id]: {
-      ...oldMessage,
-      reactions: oldMessage.reactions.filter(
-        x => !(x.emoji_name === action.emoji_name && x.user_id === action.user_id),
-      ),
-    },
-  };
+  return state.set(action.message_id, {
+    ...oldMessage,
+    reactions: oldMessage.reactions.filter(
+      x => !(x.emoji_name === action.emoji_name && x.user_id === action.user_id),
+    ),
+  });
 };
 
 const eventNewMessage = (state, action) => {
   // TODO: Optimize -- Only update if the new message belongs to at least
   // one narrow that is caught up.
-  if (state[action.message.id]) {
+  if (state.get(action.message.id)) {
     return state;
   }
-  return {
-    ...state,
-    [action.message.id]: omit(action.message, 'flags'),
-  };
+  return state.set(action.message.id, omit(action.message, 'flags'));
 };
 
 const eventSubmessage = (state, action) => {
-  const message = state[action.message_id];
+  const message = state.get(action.message_id);
   if (!message) {
     return state;
   }
-  return {
-    ...state,
-    [action.message_id]: {
-      ...message,
-      submessages: [
-        ...(message.submessages || []),
-        {
-          id: action.submessage_id,
-          message_id: action.message_id,
-          sender_id: action.sender_id,
-          msg_type: action.msg_type,
-          content: action.content,
-        },
-      ],
-    },
-  };
+  return state.set(action.message_id, {
+    ...message,
+    submessages: [
+      ...(message.submessages || []),
+      {
+        id: action.submessage_id,
+        message_id: action.message_id,
+        sender_id: action.sender_id,
+        msg_type: action.msg_type,
+        content: action.content,
+      },
+    ],
+  });
 };
 
 const eventMessageDelete = (state, action) => {
-  if (action.messageIds.every(messageId => !state[messageId])) {
+  if (action.messageIds.every(messageId => !state.get(messageId))) {
     return state;
   }
-  return omit(state, action.messageIds);
+  return state.deleteAll(action.messageIds);
```

Comment:
This can be simplified to just the `deleteAll` call. (I'm maybe 95% confident that Immutable will indeed return the identical value if none of the keys are present -- but it'd be nice to make sure there's a `messagesReducer` test confirming that.) Simpler and a bit more efficient, too.

Best as a separate followup commit, as it's nice that this commit is such a straightforward translation.